skbio/io/format/stockholm.py

```diff
@@ -695,64 +695,69 @@ def _tabular_msa_to_stockholm(obj, fh):
-                                           % gf_feature_data)
-            for ref_num, dictionary in enumerate(gf_feature_data, start=1):
-                if not isinstance(dictionary, dict):
-                    raise StockholmFormatError("Expected reference information"
-                                               " to be stored as a dictionary,"
-                                               " found reference %d stored as "
-                                               "%r."
-                                               % (ref_num,
-                                                  type(dictionary).__name__))
-                fh.write("#=GF RN [%d]\n" % ref_num)
-                for feature in dictionary:
-                    if feature not in _REFERENCE_TAGS:
-                        raise StockholmFormatError("Invalid reference tag %r "
-                                                   "found in reference dictio"
-                                                   "nary %d. Valid reference "
-                                                   "tags are: %s."
-                                                   % (feature, ref_num,
-                                                      ', '.join([
-                                                            tag for tag in
-                                                            _REFERENCE_TAGS])))
-                    fh.write("#=GF %s %s\n" % (feature,
-                                               dictionary[feature]))
-        else:
-            fh.write("#=GF %s %s\n" % (gf_feature, gf_feature_data))
+    if obj.has_metadata():
+        for gf_feature, gf_feature_data in obj.metadata.items():
+            if gf_feature == 'NH' and isinstance(gf_feature_data, dict):
+                for tree_id, tree in gf_feature_data.items():
+                    fh.write("#=GF TN %s\n" % tree_id)
+                    fh.write("#=GF NH %s\n" % tree)
+            elif gf_feature == 'RN':
+                if not isinstance(gf_feature_data, list):
+                    raise StockholmFormatError(
+                        "Expected 'RN' to contain a list of reference "
+                        "dictionaries, got %r." % gf_feature_data)
+
+                for ref_num, dictionary in enumerate(gf_feature_data, start=1):
+                    if not isinstance(dictionary, dict):
+                        raise StockholmFormatError(
+                            "Expected reference information to be stored as a "
+                            "dictionary, found reference %d stored as %r." %
+                            (ref_num, type(dictionary).__name__))
+
+                    fh.write("#=GF RN [%d]\n" % ref_num)
+                    for feature in dictionary:
+                        if feature not in _REFERENCE_TAGS:
+                            formatted_reference_tags = ', '.join(
+                                [tag for tag in _REFERENCE_TAGS])
```

Comment:
This could be a generator comprehension (sorry I know you were just re-flowing what was there)